### ISSUE TITLE
Words with number boundaries incorrectly treated as Links 

### ISSUE THREAD

**Issue by neSpecc** [nfrasser/linkifyjs#493]
When using Linkify.js to detect links in text, words with numbers as boundaries are incorrectly highlighted as links. For example, in the string "somefile.mp4", Linkify.js interprets "somefile.mp" as a link, omitting the final 4 character.

### Actual behavior

```js
linkifyHtml('some string with somefile.mp4 token') // "some string with <a href="somefile.mp">somefile.mp</a>4 token"
```

### Expected behavior

Link should not be parsed
```js
linkifyHtml('some string with somefile.mp4 token') // "some string with somefile.mp4 token"
```

**Comment by lattaai11** [comment]
Hello, I tried to solve the issue.

This is what I did:

Update the tokenize function in linkify.js to add a check for file extensions with numbers. This will prevent treating strings like 'somefile.mp4' as links.

You can review changes in this commit: https://github.com/lattaai11/nfrasser-linkifyjs-493/commit/89f7e101068839336870b980aa33d1f1629f5e01.


> [!CAUTION]
> Disclaimer: The commit was created by Latta AI and you should never copy paste this code before you check the correctness of generated code. Solution might not be complete, you should use this code as an inspiration only.

---
This issue was tried to solve for free by Latta AI - https://latta.ai/ourmission

If you no longer want Latta AI to attempt solving issues on your repository, you can block this account.

**Comment by lattaai11** [comment]
Hello, I tried to solve the issue.

This is what I did:

Update the scanner logic to prevent misidentification of file extensions containing numbers as links. This change will ensure that strings like 'somefile.mp4' are not parsed as links.

You can review changes in this commit: https://github.com/lattaai11/nfrasser-linkifyjs-493/commit/b8afd5c30bff731c6907d760b084cd57e9d497ed.


> [!CAUTION]
> Disclaimer: The commit was created by Latta AI and you should never copy paste this code before you check the correctness of generated code. Solution might not be complete, you should use this code as an inspiration only.

---
This issue was tried to solve for free by Latta AI - https://latta.ai/ourmission

If you no longer want Latta AI to attempt solving issues on your repository, you can block this account.

**Comment by nfrasser** [comment]
@neSpecc the fix for this is now available in the latest v4.2.0 release, thanks for reporting!
